### ISSUE TITLE
problems with detector, plotting and obtaining FL data

### ISSUE THREAD

**Issue by DavidMF123** [gcharvin/DetecDiv#11]
in the detector, after making the groups and extract divisions & RLS, RLS and division data cna be ploted and exported (although interdivision times are somehow sketchy to get from that file). however, with the FL signal i just can do one type of plot. if i select 'generations', and in the command line options select any of the FL options with the mask, it works with no bugs (by the way, the x axis is limited by the shortest mean RLS in all the groups, i think it should be limited by the longest instead). however, the same plot is made either if i choose to align tracks at birth SEP or death, which is weird.
moreover, if with the same options i choose the other option in the command line step (plot division time along RLS), i get the following error message/bug:
Which signal type? (Default: 1)1 - divDuration; 2 - onemask; 1
Unable to perform assignment because the size of the left side is 1-by-50 and the size of the right side is 1-by-19.

Error in plotSignal2 (line 290)
            data{cond}(r,:)=obj2{cond}{r,1}*timefactor;

Error in detector/PlotSignalExportButtonPushed (line 1003)
                plotSignal2(rlstot,arg{:});

Error in appdesigner.internal.service.AppManagementService/executeCallback (line 138)
                callback(appOrUserComponent, event);

Error in matlab.apps.AppBase>@(source,event)executeCallback(appdesigner.internal.service.AppManagementService.instance(),app,callback,requiresEventData,event) (line 63)
            newCallback = @(source, event)executeCallback(appdesigner.internal.service.AppManagementService.instance(), ...
 
Error using matlab.ui.control.internal.controller.ComponentController/executeUserCallback
Error while evaluating Button PrivateButtonPushedFcn.

then, if instead of choosing 'generations' i choose 'time', after selecting the command line parameters, i get the following:
Which signal type? (Default: 1)1 - onemask; 1
Which classi used for extracting signal? (Default: 1)1 - from_results_cellseg60X_1; 1
Which type of signal? (Default: 1)1 - volume; 2 - totalfluo; 3 - meanfluo; 3
Which channel ? (Default: 1)11
Index exceeds the number of array elements. Index must not exceed 30.

Error in plotSignal2 (line 294)
            sdatavector=[rlstmp(r).frameBirth,rlstmp(r).frameEnd];

Error in detector/PlotSignalExportButtonPushed (line 1003)
                plotSignal2(rlstot,arg{:});

Error in appdesigner.internal.service.AppManagementService/executeCallback (line 138)
                callback(appOrUserComponent, event);

Error in matlab.apps.AppBase>@(source,event)executeCallback(appdesigner.internal.service.AppManagementService.instance(),app,callback,requiresEventData,event) (line 63)
            newCallback = @(source, event)executeCallback(appdesigner.internal.service.AppManagementService.instance(), ...
 
Error using matlab.ui.control.internal.controller.ComponentController/executeUserCallback
Error while evaluating Button PrivateButtonPushedFcn.

finally, even in the case for which the FL plot along RLS is made, i dont see how i can obtain the data  in an spreadsheet or whatever, so i can create the plots at my style or whatever. there is a 'plot' button, but no an 'export' button....

so in summary, there are this 2 bugs when trying to plot division time across lifespan and trying to plot FL across time, the fact that no matter which aligning point i use i get the same plot (i guess it´s aligned at birth), and that there´s no way to export the FL raw data to plot it as i want (or at least i don´t know how to retrieve it).
i hope you can fix/improve this
thanks
david

**Comment by gcharvin** [comment]
@TAspert could you please look into the plotSignal2 function problem?

**Comment by gcharvin** [comment]
I close this issue, since fluo quantification and export have completely rewritten.
